vowpalwabbit/example.h

```diff
@@ -85,6 +90,9 @@ struct example : public example_predict  // core example datatype.
   bool end_pass;  // special example indicating end of pass.
   bool sorted;    // Are the features sorted or not?
   bool in_use;    // in use or not (for the parser)
+
+  // Interactions can be overriden on a per example basis by setting this pointer.
+  std::vector<std::string>* override_interactions;
```

Comment:
I believe this should just always be used so we only have one mechanism for specifying interactions.

Reply:
I agree, but the way to do it properly us quickly becoming out of scope in the timeframe we have left for CCB. Would it be acceptable to make this a follow up issue?

Reply:
I'd rather not, for two reasons.

The first is that I believe this can fixed with a single line of code in the example initializer.

The second is just general aversion to incurring additoinal technical debt.

Reply:
For the single line I assume you mean copying the command line provided interactions into this per-example store? I agree, this is easy and I'm totally happy to do that. My concerns is this incurs a copy for every example, CCB or not and regardless if the reduction stack wishes to mutate the interactions.

Reply:
No copy---just have have a pointer and declare the memory at that address not owned by the example.

Reply:
For some reason I had it in my mind that there would be several concurrent usages of such a piece of memory, but thinking about it now I don't think that's the case.

Then as long as the example puts it back how it found it we are good. I'll implement this.

Reply:
I can't quite tell if this is what you have in mind but using COW semantics seems plausibly sanest.  By default, reductions don't touch this field.  Any that do should create their own interactions vector and change the example to point to it, then restore the pointer to the default interactions when finishing.

Reply:
COW was the sort of thing I had in mind too. I think we are on the same page here.